lib/index.js

```diff
@@ -0,0 +1,101 @@
+import * as path from "path";
+import * as fs from "fs";
+
+import { watch } from "chokidar";
+import {sync as mkdirp} from "mkdirp";
+import most from "most";
+import _ from "lodash";
+
+import compile from "./compile";
+import { entries } from "./util";
+import loadAst from "./compile/modules/load-ast";
+import compileModules from "./compile/modules/compile";
+
+export default function Interlock (options) {
+  const cwd = process.cwd();
+
+  if (!options.emit || !options.emit.length) {
+    throw new Error("Must define at least one bundle.");
+  }
+
+  // TODO: validate options, or only persist specific values
+  this.options = _.defaults(options || {}, {
+    context: cwd,
+    outputPath: path.join(cwd, "dist"),
+    extensions: [".js", ".jsx", ".es6"],
+    ns: require(path.join(options.root, "./package.json")).name
+  });
+}
+
+Interlock.prototype.build = function () {
+  compile(this.options)
+    .then(this._saveBundles)
```

Comment:
It is possible that somebody would want to handle the save function themselves. It would be good to either make _saveBundles pluggable or just allow somebody to turn it off and return the compilation after build.